Userland/Libraries/LibJS/Runtime/Object.cpp

```diff
@@ -1155,6 +1155,34 @@ ThrowCompletionOr<Object*> Object::define_properties(Value properties)
     return this;
 }
 
+// 14.7.5.9 EnumerateObjectProperties ( O ), https://tc39.es/ecma262/#sec-enumerate-object-properties
+Optional<Completion> Object::enumerate_object_properties(Function<Optional<Completion>(Value)> callback) const
+{
+    HashTable<FlyString> visited;
+
+    const auto* target = this;
+    while (target) {
+        auto own_keys = TRY(target->internal_own_property_keys());
+        for (auto& key : own_keys) {
+            if (!key.is_string())
+                continue;
+            FlyString property_key = TRY(key.to_string(global_object()));
```

Comment:
Should you use `to_string` here? You have already checked it is a string above, and `to_string` could trigger side effects (never in this case) but using `as_string` might communicate that more clearly?

Reply:
Absolutely, good catch